### DISCUSSION TITLE
Question about useMediaQuery

### DISCUSSION THREAD

**Discussion by vip30** [react-hookz/web#931] [Q&A] [answered]
We're using `useMediaQuery` for getting the type of the current viewport, in the past ([react-use](https://github.com/streamich/react-use/blob/master/src/useMedia.ts#L25)) it can retrieve the initial state from the global variables but now the value is undefined in the initial and set after useEffect. It makes our layout getting shift.

Would like to ask the possibility to get the initial state from the global value before effect and what is purpose of this design (Media value undefined in initial)?

**Comment by xobotyi** [comment, marked as answer]
It is possible but lead to gaisenbug when in some cases state is not retrieved by no reason.

The reason for now - proper work during SSR. I have in plans to watch carefully what was the reason and attempt to fix the issue, but for now i have to leave it like that
